packages/components/tag/src/Tag.tsx

```diff
@@ -12,6 +12,21 @@ export interface TagProps
   asChild?: boolean
 }
 
+interface FilteredDesignIntent<
+  Design extends TagProps['design'],
+  K extends TagStylesProps['intent'] | never = never,
+> {
+  design?: Design
+  intent?: Exclude<TagStylesProps['intent'], K>
+}
+
+type ValidTagDesignIntent =
+  | FilteredDesignIntent<'tinted', 'surface'>
+  | FilteredDesignIntent<'outlined', 'surface'>
+  | FilteredDesignIntent<'filled'>
```

Comment:
What is being done here is to prevent unwanted combinations.
The surface **intent** can only be applied if the design is set to **filled**.
We cannot have the following:
```tsx
<Tag design="outlined" intent="surface">
```